src/Http/Routing.Abstractions/src/RoutingHttpContextExtensions.cs

```diff
@@ -23,8 +24,8 @@ public static RouteData GetRouteData(this HttpContext httpContext)
                 throw new ArgumentNullException(nameof(httpContext));
             }
 
-            var routingFeature = httpContext.Features[typeof(IRoutingFeature)] as IRoutingFeature;
-            return routingFeature?.RouteData;
+            var routingFeature = httpContext.Features.Get<IRoutingFeature>();
+            return routingFeature?.RouteData ?? new RouteData(httpContext.Request.RouteValues);
```

Comment:
Should this assign `routingFeature.RouteData`?

Reply:
I dunno feels wrong and side effecty. I don’t want to create a feature here especially since it really isn’t faithful part of the future. Can you think of a downside to not setting the feature?

Reply:
I guess I'm wondering about the case where you have a routing feature but route data is null. This method would return (but not store) a new RouteData object.

Reply:
> I guess I'm wondering about the case where you have a routing feature but route data is null. This method would return (but not store) a new RouteData object.

When would that happen?

Reply:
I’ll make the change but I’m not sure when this would happen...